moveos/moveos-object-runtime/src/runtime_object.rs

```diff
@@ -239,6 +239,72 @@ impl RuntimeObject {
         })
     }
 
+    /// Retrieves a field of the object from the state store.
+    pub fn retrieve_field(
+        &self,
+        layout_loader: &dyn TypeLayoutLoader,
+        resolver: &dyn StatelessResolver,
+        field_key: FieldKey,
+    ) -> PartialVMResult<(RuntimeObject, Option<Option<NumBytes>>)> {
+        let state_root = self.state_root()?;
+        let field_obj_id = self.id().child_id(field_key);
+
+        match resolver
+            .get_field_at(state_root, &field_key)
+            .map_err(|err| {
+                partial_extension_error(format!("remote object resolver failure: {}", err))
+            })? {
```

Comment:
We should `get_loaded_field` first because the field may be changed by this transaction, then get the field via the resolver from db.

Reply:
OK, I will fix it.

Reply:
Since the RuntimeObject returned by get_loaded_field is imported and does not support Clone, I moved this logic to get_field and scan_fields, and made retrieve_field and list_fields private.